models/GroupsService.php

```diff
@@ -15,37 +15,58 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  *
- * Copyright (c) 2002-2008 (original work) Public Research Centre Henri Tudor & University of Luxembourg (under the project TAO & TAO2);
- *               2008-2010 (update and modification) Deutsche Institut für Internationale Pädagogische Forschung (under the project TAO-TRANSFER);
- *               2009-2012 (update and modification) Public Research Centre Henri Tudor (under the project TAO-SUSTAIN & TAO-DEV);
- *               2013-2014 (update and modification) Open Assessment Technologies SA
+ * Copyright (c) 2002-2008 (original work) Public Research Centre Henri Tudor & University of Luxembourg
+ *                         (under the project TAO & TAO2);
+ *               2008-2010 (update and modification) Deutsche Institut für Internationale Pädagogische Forschung
+ *                         (under the project TAO-TRANSFER);
+ *               2009-2012 (update and modification) Public Research Centre Henri Tudor
+ *                         (under the project TAO-SUSTAIN & TAO-DEV);
+ *               2013-2023 (update and modification) Open Assessment Technologies SA
  */
 
 namespace oat\taoGroups\models;
 
-use \core_kernel_classes_Class;
-use \core_kernel_classes_Property;
-use \core_kernel_classes_Resource;
+use oat\generis\model\data\Ontology;
 use oat\taoTestTaker\models\TestTakerService;
 use oat\oatbox\user\User;
 use oat\tao\model\OntologyClassService;
+use common_Exception;
+use common_exception_Error;
+use core_kernel_classes_Class;
+use core_kernel_classes_Resource;
+use League\Flysystem\FileExistsException;
 
 /**
  * Service methods to manage the Groups business models using the RDF API.
  *
  * @access public
  * @author Joel Bout, <joel.bout@tudor.lu>
  * @package taoGroups
-
  */
 class GroupsService extends OntologyClassService
 {
-    const CLASS_URI = 'http://www.tao.lu/Ontologies/TAOGroup.rdf#Group';
+    public const CLASS_URI = 'http://www.tao.lu/Ontologies/TAOGroup.rdf#Group';
+
+    public const PROPERTY_MEMBERS_URI = 'http://www.tao.lu/Ontologies/TAOGroup.rdf#member';
+
+    private TestTakerService $testTakerService;
+
+    public function __construct(
+        $options = [],
+        ?TestTakerService $testTakerService = null,
+        ?Ontology $ontology = null
+    ) {
```

Comment:
Do we really need TestTakerService and Ontology in constructor?

Reply:
I prefer using constructor injection, but initially this was indeed using the methods from `ConfigurableService`. I will rollback changes from 40bc9e4d8388cc4bde8bbcb1d6e120a0875761ec and keep using the previous way for getting services.

Reply:
I also prefer to use `constructor injection` and we should not use `ConfigurableService` and instead use DI container. The thing is that we are not using DI here, we are using `ConfigurableService`, so it is better to remain using it instead of create t third way just for testing purposes (which is not necessary).

Reply:
Rolled back changes to use constructor injection in cb5be793d4c39badf9e7fb89109b0f77246b9077.